sdk/how-to-guides/leadership.md

```diff
@@ -0,0 +1,148 @@
+    secret.set_content({"username": "user", "password": "pass"})
+```
+
+Note that Juju guarantees leadership for only 30 seconds after a `leader-elected`
+event or an `is-leader` check. If the charm code may run longer, then extra
+`is_leader()` calls should be made to ensure that the unit is still the leader.
+
+<a href="#heading--test-leadership-management"><h2 id="heading--test-leadership-management">Test leadership management</h2></a>
+
+> See first: [Get started with charm testing](/t/6894)
+
+You'll want to add three levels of tests:
+
+- [Write unit tests](#heading--write-unit-tests)
+- [Write scenario tests](#heading--write-scenario-tests)
+- [Write integration tests](#heading--write-integration-tests)
+
+<a href="#heading--write-unit-tests"><h3 id="heading--write-unit-tests">Write unit tests</h3></a>
+
+> See first: [How to write unit tests for a charm](/t/4461)
+
+When using Harness for unit tests, use the `set_leader()` method to control whether the unit is the leader. For example, to verify that leadership change is handled correctly:
+
+```python
+@pytest.fixture()
+def harness():
+    yield ops.testing.Harness(MyCharm)
+    harness.cleanup()
+
+
+def test_new_leader(harness):
+    # Before the test, the unit is not leader.
+    harness.set_leader(False)
+    harness.begin()
+    # Simulate Juju electing the unit as leader.
+    harness.set_leader(True)
+    # Assert that it was handled correctly.
+    assert ...
+
+
+def test_leader_sets_secrets(harness):
+    # The unit is the leader throughout the test, and no leader-elected event
+    # is emitted.
+    harness.set_leader(True)
+    harness.begin()
+    secret_id = harness.add_model_secret(APP_NAME, content={"secret": "sssh"})
+    harness.update_config(secret_option=secret_id)
+    # Assert that the config-changed handler set additional secret metadata:
+    assert ...
+```
+
+> See more: [`ops.testing.Harness.set_leader`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness.set_leader)
+
+<a href="#heading--write-scenario-tests"><h3 id="heading--write-scenario-tests">Write scenario tests</h3></a>
+
+> See first: [How to write scenario tests for a charm](/t/10585)
+
+When using Scenario for unit tests, pass the leadership status to the `State`. For example:
+
+```python
+class MyCharm(ops.CharmBase):
+    def __init__(self, framework):
+        super().__init__(framework)
+        framework.observe(self.on.start, self._on_start)
+
+    def _on_start(self, _):
+        if self.unit.is_leader():
+            self.unit.status = ops.ActiveStatus('I rule')
+        else:
+            self.unit.status = ops.ActiveStatus('I am ruled')
+
+
+@pytest.mark.parametrize('leader', (True, False))
+def test_status_leader(leader):
+    ctx = scenario.Context(MyCharm, meta={"name": "foo"})
+    out = ctx.run('start', scenario.State(leader=leader))
+    assert out.unit_status == ops.ActiveStatus('I rule' if leader else 'I am ruled')
+```
+
```

Comment:
> See more missing. Is it because we don't have Scenario docs yet?

Essentially, yes. In other guides we've linked to a section in the README, but there isn't one specifically about leadership (it is mentioned in other places).